### PR TITLE
Changes to documentation & core.py

### PR DESCRIPTION
**Pull request by lewis-morris**
## Updates

- `core.py` so it will accept `path parameters` descriptions. This is added to the view's docstring. It's a bit hacky and I'm sure you can think of a better way of doing this. But I wanted the feature.

- `core.py` also now reads the docstring from custom decorators so you can append the "description" of that endpoint with extra information from each decorator   i.e   extra info from @paginated_response

- documentation addition,  you can also feel free to change if you want, but others could benefit from this.

So many commits to the docs but I've not really used rst before or sphynx before so had to play about with it.

### CODE REVIEW

File: docs/documentation_comments.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+A small example is shown below.
+
+.. code-block:: python
+
+  class FilterUserSchema(ma.Schema): 
+      class Meta:
+          ordered = True
+      email = ma.String()
+      email.metadata["description"] = "Accepts operators [GT] [LT] [GTE] [LTE]"
```
Comment (miguelgrinberg) [inline docs/documentation_comments.rst:74]: This can be done in a single line:

    email = ma.String(description='some description text here')

**Comment by miguelgrinberg** [comment]
Hey, thanks for the updates. I have to review the changes you've made in core.py, but the docs improvements are a great addition that I definitely can use.

Note that I will not be able to merge all these things that you have changed together, as I prefer to not batch several changes in a single commit, as that makes reviewing the code in the future much harder.

File: src/apifairy/core.py [hunk 2]
```diff
@@ -147,14 +146,10 @@ def resolver(schema):
...
         ma_plugin.converter.field_mapping[fields.AbsoluteURLFor] = \
             ('string', 'url')
-        if sqla is not None:  # pragma: no cover
+        if sqla is not None:
             ma_plugin.converter.field_mapping[sqla.HyperlinkRelated] = \
                 ('string', 'url')
 
-        # configure FileField
-        ma_plugin.converter.field_mapping[apifairy_fields.FileField] = \
-            ('string', 'binary')
```
Comment (miguelgrinberg) [inline src/apifairy/core.py:156]: I think you have some merging issues in this PR, because I don't see why you would delete this. My guess is that you are working off of an older version of this code that doesn't have this change.

**Comment by miguelgrinberg** [comment]
Okay, here is some feedback:

- For documenting path parameters, I do like the idea of doing it, but creating a custom format within the docstring to do this is not great, especially because there are already standards for this, like the format that Sphinx uses. I was thinking about this, and came up with the idea of using annotations for this, which works well as long as the project doesn't use annotations for type hints. Example:

    ```python
    @users.route('/users/<int:id>', methods=['GET'])
    @authenticate(token_auth)
    @response(user_schema)
    @other_responses({404: 'User not found'})
    def get(id: 'The id of the user'):
        """Retrieve a user by id"""
        return db.session.get(User, id) or abort(404)
    ```

- Appending comments from decorators is a strange idea, because you really have no way to know which decorators you may want to take docs from or which you wouldn't. If I understand your change correctly you are only looking at decorators that wrap the ``arguments`` decorator from this page, and only if the `query` location is used. Any decorators that do something outside of query string arguments would be ignored. So I'm going to pass on this change, as I don't see how it can work in a more generic way.

So basically what I'm going to do is commit the annotation change separately, and from this PR I'm going to use the new documentation page that you created and complete it with the remaining documentation options that you have not included.

Thanks!

**Comment by lewis-morris** [comment]
> Okay, here is some feedback:
> 
> * For documenting path parameters, I do like the idea of doing it, but creating a custom format within the docstring to do this is not great, especially because there are already standards for this, like the format that Sphinx uses. I was thinking about this, and came up with the idea of using annotations for this, which works well as long as the project doesn't use annotations for type hints. Example:
>   ```python
>   @users.route('/users/<int:id>', methods=['GET'])
>   @authenticate(token_auth)
>   @response(user_schema)
>   @other_responses({404: 'User not found'})
>   def get(id: 'The id of the user'):
>       """Retrieve a user by id"""
>       return db.session.get(User, id) or abort(404)
>   ```
> * Appending comments from decorators is a strange idea, because you really have no way to know which decorators you may want to take docs from or which you wouldn't. If I understand your change correctly you are only looking at decorators that wrap the `arguments` decorator from this package, and only if the `query` location is used. Any decorators that do something outside of query string arguments would be ignored. So I'm going to pass on this change, as I don't see how it can work in a more generic way.
> 
> So basically what I'm going to do is commit the annotation change separately, and from this PR I'm going to use the new documentation page that you created and complete it with the remaining documentation options that you have not included.
> 
> Thanks!

Anything you use is fine by me.

**Comment by lewis-morris** [comment]
> Okay, here is some feedback:
> 
> * For documenting path parameters, I do like the idea of doing it, but creating a custom format within the docstring to do this is not great, especially because there are already standards for this, like the format that Sphinx uses. I was thinking about this, and came up with the idea of using annotations for this, which works well as long as the project doesn't use annotations for type hints. Example:
>   ```python
>   @users.route('/users/<int:id>', methods=['GET'])
>   @authenticate(token_auth)
>   @response(user_schema)
>   @other_responses({404: 'User not found'})
>   def get(id: 'The id of the user'):
>       """Retrieve a user by id"""
>       return db.session.get(User, id) or abort(404)
>   ```
> * Appending comments from decorators is a strange idea, because you really have no way to know which decorators you may want to take docs from or which you wouldn't. If I understand your change correctly you are only looking at decorators that wrap the `arguments` decorator from this package, and only if the `query` location is used. Any decorators that do something outside of query string arguments would be ignored. So I'm going to pass on this change, as I don't see how it can work in a more generic way.
> 
> So basically what I'm going to do is commit the annotation change separately, and from this PR I'm going to use the new documentation page that you created and complete it with the remaining documentation options that you have not included.
> 
> Thanks!

The idea of adding documentation to the decorator was that you could add, for instance, a "filter" decorator that accepts any schema you like, but having generic instructions on its use that would relate to all schemas. 

Maybe It only fits my use case, but it's working for me.

**Comment by miguelgrinberg** [comment]
Added new documentation here: https://apifairy.readthedocs.io/en/latest/guide.html.

Thanks for your help!
